src/app/core/uri-retriever.service.ts

```diff
@@ -0,0 +1,308 @@
+   * @param properties the properties (i.e. children, parents, etc) that have been requested
+   */
+  getLearningObjectResources(
+    params: { author?: string, name?: string, id?: string },
+    properties?: string[]
+  ) {
+    try {
+      const route = this.setRoute(params);
+
+      const responses: Subject<any> = new Subject();
+      const end = new Subject();
+
+      this.http.get<LearningObject>(route).pipe(
+        retry(3),
+        tap(entity => {
+          const uris = Object.assign(entity.resourceUris);
+          let completed = 0;
+          Object.keys(uris).map(key => {
+            if (!properties || properties.includes(key)) {
+              this.fetchUri(uris[key]).subscribe(value => {
+                responses.next({ requestKey: key, value });
+                if (++completed === properties.length || completed === uris.length) {
+                  responses.complete();
+                  end.next();
+                  end.complete();
+                }
+              });
+            }
+          });
+
+          responses.next(new LearningObject(entity));
+        }),
+        takeUntil(end),
+        catchError(this.handleError),
+      ).subscribe();
+
+      return responses;
+    } catch (err) {
+      throw err;
+    }
+  }
+
+
+  //////////////////////////
+  // INDIVIDUAL RESOURCES //
+  /////////////////////////
+
+  /**
+   * Retrieves the Learning Object metadata
+   * @params author is the username of the author
+   * @params learningObjectName is the name of the learning object
+   */
+  getLearningObjectSummary(author: string, learningObjectName: string): Observable<any> {
+    return this.http.get<LearningObject>(
+      PUBLIC_LEARNING_OBJECT_ROUTES.GET_PUBLIC_LEARNING_OBJECT(
+        author, learningObjectName
+        ))
+        .pipe(
+          retry(3),
+          catchError(this.handleError)
+        );
+  }
+
+  /**
+   * Retrieves the Learning Object outcomes
+   * @param uri this is the uri that should be hit to get the objects outcomes
+   */
+  getLearningObjectOutcomes(uri: string): Observable<LearningOutcome[]> {
+    return this.http.get<LearningOutcome[]>(uri).pipe(retry(3), catchError(this.handleError));
+  }
+
+  /**
+   * Retrieves the Learning Object children
+   * @param uri this is the uri that should be hit to get the objects children
+   */
+  getLearningObjectChildren(params: {uri: string, unreleased?: boolean}): Observable<LearningObject[]> {
+    if (params.unreleased) {
+      return this.http.get<LearningObject[]>(params.uri, { withCredentials: true })
+      .pipe(
+        retry(3),
+        catchError(this.handleError),
+        take(1),
+        /* TODO: Remove this.
+         * It is a stopcap until the service stops returning unreleased.
+         * More info at https://github.com/Cyber4All/learning-object-service/pull/282
+         */
+        map(children => children.filter(child => child.status !== 'unreleased')),
```

Comment:
This can be removed since https://github.com/Cyber4All/learning-object-service/pull/282 is merged.